app/controllers/AccountManagement.scala

```diff
@@ -351,22 +354,29 @@ object AccountManagement extends Controller with ContextLogging with CatalogProv
     implicit val tpBackend = resolution.backend
     val eventualMaybeSubscription = SessionSubscription.subscriptionFromRequest
     val errorCodes = errorCode.toSeq.flatMap(_.split(',').map(_.trim)).filterNot(_.isEmpty).toSet
+    val futureMaybeMaybeEmail = (for {
+      authUser <- OptionT(Future.successful(authenticatedUserFor(request)))
+      idUser <- OptionT(IdentityService.userLookupByCredentials(authUser.credentials))
+    } yield Option(idUser.primaryEmailAddress)).run.recover{case _ => None}
```

Comment:
is `idUser.primaryEmailAddress` likely to be null? the Option constructor is to handle null, otherwise use `Some(...)`

Reply:
also might be worth logging the error when you recover (at least info/warn), otherwise it will fail silently

Reply:
My bad.